### DISCUSSION TITLE
Channel Model without OFDM waveform

### DISCUSSION THREAD

**Discussion by aritra-2506** [NVlabs/sionna#93] [Q&A] [answered]
I want to use channel models like TDL, CDL or UMi without using the OFDM waveform (resource grid). Is this possible in SIONNA? I do not see an option.

**Comment by jhoydis** [comment, marked as answer]
Hi,

There are various ways in which you can use these channel models.

1. You can create an instance of the channel model, e.g., [UMi](https://nvlabs.github.io/sionna/api/channel.wireless.html#sionna.channel.tr38901.UMi), and generate batches of impulse responses, i.e., path delays and complex gains. Then you are free to do with this whatever you want.
2. You can create a discrete frequency-domain representation of the channel impulse response generated in step 1.  by applying the [cir_to_ofdm_channel](https://nvlabs.github.io/sionna/api/channel.wireless.html#sionna.channel.cir_to_ofdm_channel) function.
3. You can create a discrete complex-baseband time-domain representation of the channel impulse response generated in step 1. by applying the [cir_to_time_channel](https://nvlabs.github.io/sionna/api/channel.wireless.html#sionna.channel.cir_to_time_channel) function.

The functions [ApplyTimeChannel](https://nvlabs.github.io/sionna/api/channel.wireless.html#sionna.channel.ApplyTimeChannel) and [ApplyOFDMChannel](https://nvlabs.github.io/sionna/api/channel.wireless.html#sionna.channel.ApplyOFDMChannel) can be used to compute channel outputs for a provided input signal using the results from step 2. or step 3.

None of these functions rely on a resource grid and you can use them in any way you want. You could also use them for single-carrier modulations, e.g., by using some filters from the [signal](https://nvlabs.github.io/sionna/api/signal.html) module.

Hope that helps.

**Reply by aritra-2506** [reply]
Thanks a lot Mr. Hoydis. It will surely help.
